stable_baselines3/common/policies.py

```diff
@@ -251,7 +251,7 @@ def obs_to_tensor(self, observation: Union[np.ndarray, Dict[str, np.ndarray]]) -
         return observation, vectorized_env
 
 
-class BasePolicy(BaseModel):
+class BasePolicy(BaseModel, ABC):
```

Comment:
because it doesn't contain any unimplemented method anymore?

Reply:
Yes

Reply:
well, it does contain unimplemented `forward()`...

Reply:
but pytorch doesn't define it like that because of typing... we could just do `# noqa: B024` in that case.

Reply:
Inheriting `nn.Module` without implementing `forward` is a problem only if the method is intended to be instantiated (
[PEP 3119](https://peps.python.org/pep-3119/#the-abc-module-an-abc-support-framework)).
Since `BaseModel` is not intended to be instanciated, I think the right way is the following:


```python
from abc import ABC, abstractmethod


class Module:  # Doesn't inherit from ABC, from some reason
    def forward(self):
        raise NotImplementedError()


class BaseModel(Module):
    ...


class BasePolicy(BaseModel, ABC):
    @abstractmethod
    def _predict(self):
        ...

```

Note that even if we try to instantiate `BaseModel`, an error is raised. This error comes directly from torch and not from abc, but I think it is out of the scope of SB3.

Reply:
To clarify my previous post, even if `nn.Module` was an abstract class inheriting from `ABC`, it would be the same thing:

```python
from abc import ABC, abstractmethod


class Module(ABC):
    @abstractmethod
    def forward(self):
        ...


class BaseModel(Module):
    ...


class BasePolicy(BaseModel, ABC):
    @abstractmethod
    def _predict(self):
        ...
```

Reply:
I guess removing `ABC` completely is not an option neither?

Reply:
We could find a sneaky way, like the one used in `nn.Module`. But what for?